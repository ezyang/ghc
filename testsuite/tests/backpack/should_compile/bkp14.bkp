package p where
    signature H where
        data T
        f :: T
package q where
    module F where
        data T = T
        f = T
    module H(T, f) where
        import F
    include p
