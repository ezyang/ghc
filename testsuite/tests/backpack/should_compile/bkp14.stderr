[1 of 3] Processing p
unit p(hole:H, hole:Y)
  [1 of 3] Compiling H[sig]           ( p/H.hsig, nothing )
  [2 of 3] Compiling Y[sig]           ( p/Y.hsig, nothing )
  [3 of 3] Compiling M                ( p/M.hs, nothing )
[2 of 3] Processing impl
  Instantiating impl
unit impl
  [1 of 2] Compiling F                ( impl/F.hs, bkp14-out/impl/F.o )
  [2 of 2] Compiling H                ( impl/H.hs, bkp14-out/impl/H.o )
[3 of 3] Processing q
unit q(hole:Y)
- include impl [(H, impl:H), (F, impl:F)]
- include p(impl:H, hole:Y) [(M, p(impl:H, hole:Y):M)]
  Checking impl
  [1 of 2] Compiling Y[sig]           ( q/Y.hsig, nothing )
  [2 of 2] Compiling X                ( q/X.hs, nothing )
